### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by DeVuDeveloper**
Hi,
- created new files with classes Classroom and Book and Rental
- created relationship many-many between Person and book through class Rental
- no linter errors
- required changes made
- required changes have done.. changed instance of objects and method from rental to rentals when necessary

### CODE REVIEW

File: classes/rental.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Rental
+  attr_accessor :date
+
+  def initialize(date)
+    @date = date
+  end
+end
```
Comment (EricMbouwe) [inline classes/rental.rb]: - [ ] A rental belongs to a **book** and also belongs to a **person**. Getters and Setters for @book and @person instance variables should be present
- [ ] The rental should be adding itself to the @book achieving the 'belongs-to' association. hint: `book.rentals << self`

File: classes/rental.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Rental
+  attr_accessor :date
+
+  def initialize(date)
+    @date = date
+  end
```
Comment (EricMbouwe) [inline classes/rental.rb]: - [ ] The constructor of Rental should be modified so Book and Person are set in it to break the many-to-many relationship. Here is an [article ](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) explaining it

File: classes/book.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Book
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+  end
```
Comment (EricMbouwe) [inline classes/book.rb]: - [ ] A book has many rentals, there is actually no association with rentals, Getters and Setters for @rentals should be present. Let's fix that

Comment (EricMbouwe) [inline classes/book.rb]: - [ ] @rentals should be initialized in the constructor as an empty array.
- [ ] A method for adding rentals should be present. As a refresher here is the [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) to the example in a previous lesson

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
### Hi @VuDej,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlight!
- Good PR ✔️
- Good commits ✔️

# Required Changes ♻️ ⛔️

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/book.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+class Book
+  attr_accessor :title, :author, :rental
```
Comment (dasileker) [inline classes/book.rb]: - [ ] As the rental can be attributed to many so `rentals` plurals is the best choice.

File: classes/book.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rental = []
+  end
+
+  def add_rental(rental)
+    @rental.push(rental)
+    rental.book = self
```
Comment (dasileker) [inline classes/book.rb]: - [ ] Please take a careful look at what my colleague said about the rental, but here is the hint `The add_rental method takes one argument which is the date for when the book was rented out as well as a Book object with information about what book it was and who rented it out on that day.
 The code attempts to initialize a new person object with an age of 18, and a name of "Bob".
 The add_rental method will create a new rental object for the book "War and Peace" on January 1st, 2019.
`

File: classes/person.rb [hunk 6]
```diff
@@ -10,6 +10,12 @@ def initialize(age, name = 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rental = []
+  end
+
+  def add_rental(rental)
+    @rental.push(rental)
+    rental.person = self
```
Comment (dasileker) [inline classes/person.rb]: - [ ] The same thing goes for the **person.rb**  here is a hint  about the class person ` The code is a class that has three attributes: name, age, and rentals.
 The initialize method is used to create an instance of the class with the given values for those attributes.
 The add_rental method creates a new rental object using the date as its key and the book as its value.
 The code creates a new class called Person.
 The code also defines the following attributes: name, age, and rentals.`

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @VuDej,** 

### **Great work on making the changes 👏** 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## **Suggested changes**
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🎉🍾✔**
Hello @VuDej 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**